src/words/FrameLoop.java

```diff
@@ -87,8 +88,17 @@ private boolean executeSingleFrame() {
 		}
 
 		// Phase 3: Listener Evaluation
-		for (WordsEventListener eventListener : environment.getEventListeners()) {
-			eventListener.execute();
+		for (Iterator<WordsEventListener> iterator = environment.getEventListeners().iterator(); iterator.hasNext();) {
+			try {
+				WordsEventListener eventListener = iterator.next();
+				boolean delete = !eventListener.execute(environment);
+				if (delete) {
+					iterator.remove();
+				}
+			} catch (WordsRuntimeException e) {
+				System.err.println(e.toString());
```

Comment:
If we directly print a runtime exception, we're not giving the user any information about where the program went wrong (only what went wrong). By here, I think this should be a WordsProgramException.